tooling/api/tsup.config.ts

```diff
@@ -0,0 +1,30 @@
+import { defineConfig } from 'tsup'
+
+export default defineConfig(() => [
+  {
+    entry: ['src/*.ts'],
+    outDir: 'dist',
+    format: ['esm', 'cjs'],
+    clean: true,
+    minify: true,
+    platform: 'browser',
+    dts: {
+      resolve: true
+    }
+  },
+  {
+    entry: { bundle: 'src/index.ts' },
+    outDir: '../../core/tauri/scripts',
+    format: ['iife'],
+    globalName: '__TAURI_IIFE__',
+    clean: false,
+    minify: true,
+    platform: 'browser',
+    dts: false,
+    // esbuild `globalName` option generates `var __TAURI_IIFE__ = (() => {})()`
+    // but we want to assign it to the `window` object, this footer takes care of that
+    footer: {
+      js: 'window.__TAURI__ = __TAURI_IIFE__'
+    }
```

Comment:
```suggestion
    globalName: '__TAURI__',
    clean: false,
    minify: true,
    platform: 'browser',
    dts: false,
```

global functions automatically gets assigned to `window` and no need to do it again.

Reply:
from MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/var
> Note that in both NodeJS [CommonJS](https://www.commonjs.org/) modules and native [ECMAScript modules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules), top-level variable declarations are scoped to the module, and are not, therefore added as properties to the global object.

So there is cases where `var` doesn't assign to the global `window` object so I make sure to assign it manually.

Reply:
I will update the comment though